common/components/ui/Identicon.tsx

```diff
@@ -1,49 +1,43 @@
-import { isValidETHAddress } from 'libs/validators';
 import React from 'react';
 import makeBlockie from 'ethereum-blockies-base64';
+import { connect } from 'react-redux';
+import { getIsValidAddressFn } from 'selectors/config';
+import { AppState } from 'reducers';
+import './Identicon.scss';
 
-interface Props {
+interface OwnProps {
   address: string;
   className?: string;
   size?: string;
 }
 
-export default function Identicon(props: Props) {
+interface StateProps {
+  isValidAddress: ReturnType<typeof getIsValidAddressFn>;
+}
+
+type Props = OwnProps & StateProps;
+
+const Identicon: React.SFC<Props> = props => {
   const size = props.size || '4rem';
-  const { address, className = '' } = props;
-  const identiconDataUrl = isValidETHAddress(address) ? makeBlockie(address) : '';
+  const { address, isValidAddress, className = '' } = props;
```

Comment:
Prefer using the supplied defaultProps property rather than supplying a default destructure value here

Reply:
Noticed this isnt part of the PR, my bad